internal/controller/proxmoxcluster_controller.go

```diff
@@ -168,6 +168,23 @@ func (r *ProxmoxClusterReconciler) reconcileNormal(ctx context.Context, clusterS
 	// If the ProxmoxCluster doesn't have our finalizer, add it.
 	ctrlutil.AddFinalizer(clusterScope.ProxmoxCluster, infrav1alpha1.ClusterFinalizer)
 
+	if clusterScope.ProxmoxCluster.Spec.ExternalManagedControlPlane {
+		switch {
```

Comment:
I don't think switch is the right construct here considering there are two cases and each tests a different variable. Why not if else?